src/core/lib/security/credentials/fake/fake_credentials.c

```diff
@@ -35,21 +35,31 @@
 
 #include <string.h>
 
-#include "src/core/lib/channel/channel_args.h"
-#include "src/core/lib/iomgr/executor.h"
-
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
 #include <grpc/support/string_util.h>
 
+#include "src/core/ext/lb_policy/grpclb/secure_naming.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/iomgr/executor.h"
+
 /* -- Fake transport security credentials. -- */
 
 static grpc_security_status fake_transport_security_create_security_connector(
```

Comment:
Since you're passing in both the expected and actual target names here, you could do the check immediately instead of waiting for the check_peer() call.  This would eliminate the need for many of the changes in security_connector.c.